grafast/grafserv/src/middleware/graphql.ts

```diff
@@ -436,16 +436,44 @@ export const makeGraphQLHandler = (
           ? parseGraphQLBody(resolvedPreset, request, await request.getBody())
           : parseGraphQLQueryParams(await request.getQueryParams());
 
-      // Apply our hooks (if any) to the body (they will mutate the body in place)
-      const hookResult =
-        hooks.callbacks.processGraphQLRequestBody != null
-          ? hooks.process("processGraphQLRequestBody", {
-              body: parsedBody,
-              request,
-            })
-          : undefined;
-      if (hookResult != null) {
-        await hookResult;
+      try {
+        // Apply our hooks (if any) to the body (they will mutate the body in place)
+        const hookResult =
+          hooks.callbacks.processGraphQLRequestBody != null
+            ? hooks.process("processGraphQLRequestBody", {
+                body: parsedBody,
+                request,
+              })
+            : undefined;
+        if (hookResult != null) {
+          await hookResult;
+        }
+      } catch (e) {
+        if (e instanceof SafeError) {
+          const payload = {
+            errors: [
+              new GraphQLError(
+                e.message,
+                null,
+                undefined,
+                undefined,
+                undefined,
+                e,
+                undefined,
+              ),
+            ],
+          };
+          return {
+            type: "graphql",
+            request,
+            dynamicOptions,
+            statusCode: e.extensions.statusCode,
+            contentType: chosenContentType,
+            payload,
+          };
+        } else {
+          throw e;
+        }
```

Comment:
Why should this error be handled differently from an error coming from `validateGraphQLBody`?

I think it would be better to remove this try/catch and instead handle sending the GraphQL error (with the correct status code) in the `catch` block that already exists below. Be sure that `maskPayload` is called as it is in the happy path.

Reply:
Yes, you are right. I'll fix it in a next commit.

Reply:
new commit